src/tests/Microsoft.PowerFx.Interpreter.Tests/UserDefinedTests.cs

```diff
@@ -24,5 +25,16 @@ public void NamedFormulaEntryTest(string script, string expression, double expec
             var result = (NumberValue)check.GetEvaluator().Eval();
             Assert.Equal(expected, result.Value);
         }
+
+        [Theory]
+        [InlineData("test2(b: Boolean): Boolean = { Set(a, b); };")]
+        [InlineData("test2(b: Boolean): Boolean = { Set(a, b); Collect(abc, { bcd: 1 }) };")]
+        [InlineData("test2(b: Boolean): Boolean = { Set(a, b); Collect(abc, { bcd: 1 }) }; num = 3;")]
+        public void UserDefinedFunctions(string script)
+        {
+            var result = UserDefinitions.Process(script, null);

```

Comment:
>UserDefinitions.Process(script, null); [](http://example.com/codeflow?start=25&length=38)

How do we differentiate UDFs that have side effects versus those that don't? For example, in Power Apps we cannot use the former in a data property, but we can use the latter

Reply:
I assume that registered UDF will have AllowSideAffect property set to true. But that doesn't exist today and needs to be added.